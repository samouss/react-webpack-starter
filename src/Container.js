import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Container extends Component {

  constructor(props) {
    super(props);

    this.state = {
      message: '<Container />',
    };
  }

  render() {
    return (
      <div className="container">
        <h1>React starter, {this.props.title}</h1>

        Hello from {this.state.message} !!
      </div>
    );
  }

}

Container.propTypes = {
  title: PropTypes.string.isRequired,
};

export default Container;
