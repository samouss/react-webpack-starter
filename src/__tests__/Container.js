import React from 'react';
import { shallow } from 'enzyme';
import Container from '../Container';

describe('sample', () => {
  it('should pass', () => {
    const expectation = true;
    const actual = true;

    expect(actual).toEqual(expectation);
  });

  it('should pass too', () => {
    const fn = jest.fn();

    fn();

    expect(fn).toHaveBeenCalled();
  });

  it('should pass with JSX', () => {
    const props = {
      title: 'My React App',
    };

    const component = shallow(
      <Container
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
