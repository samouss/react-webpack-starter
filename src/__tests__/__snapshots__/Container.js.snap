// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sample should pass with JSX 1`] = `
<div
  className="container"
>
  <h1>
    React starter, 
    My React App
  </h1>
  Hello from 
  &lt;Container /&gt;
   !!
</div>
`;
